docs/support/tutorials/snakemake-puhti.md

```diff
@@ -0,0 +1,298 @@
+
+Please make sure that you have a [user account at CSC](../../accounts/how-to-create-new-user-account.md) and are a member of a project which [has access to the Puhti service](../../accounts/how-to-add-service-access-for-project.md) before start running workflows on Puhti. Please avoid launching heavy Snakemake workflow jobs on **login nodes** and use interactive or batch jobs instead. More information on using interactive jobs can be found on 
+[CSC documentation](../../computing/running/interactive-usage.md).
+
+Let's use the following toy Snakemake script, Snakefile (with a capital S and no file extension), for the illustration:
+
+```bash
+
+rule all:
+        input: "CAPITAL_CASE.txt"
+
+rule say_hello:
+        output: "smaller_case.txt"
+        shell:
+                """
+                echo "hello-world greetings from csc to snakemake community !" > smaller_case.txt
+                """
+rule capitalise:
+        input: "smaller_case.txt"
+        output: "CAPITAL_CASE.txt"
+        shell:
+                """
+                tr '[:lower:]' '[:upper:]' < {input} > {output}
+                """
+```
+As snakemake is installed as a module on Puhti, one can simply load it without needing to install it separately. Here is a bash script for running the above toy example:
+```bash
+module load snakemake/7.17.1
+snakemake --help   #  to get information on more options.
+
+# The following command shows how to run a snakemake workflow on a cluster using slurm executor
+snakemake -s Snakefile \ # the Snakefile is the default file name; no need specify with -s flag
+--jobs 1  \     # this will execute up to 3 tasks in parallel)       
+--latency-wait 60 \  # snakemake to wait up to 60 seconds after a job completes for the output files to become available.
+--cluster "sbatch --time 10  --account=project_xxx --job-name=hello-world --tasks-per-node=1 --cpus-per-task=1 
+--mem-per-cpu=4000 --partition test"
+# cluster option to execute the workflow on slurm
+```
+Finally, you can run the workflow in the bash script (e.g., run_snakemak.sh) by submitting in the job in the interactive node:
+```bash
+sinteractive --cores 2 --mem 10000 # type this command on login node to start an interactive session with 2 CPU cores and 10 Gb of memory, after providing your project when prompted, you can run the bash script like this:
+bash run_snakemake.sh   # run the workflow
+```
+!!! note
+    Please note that **cluster** configurations are no longer supported on recent versions of snakemake. Please consider using   [profiles](https://snakemake.readthedocs.io/en/stable/executing/cli.html#profiles) instead.
```

Comment:
Maybe we can link this as well: https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/slurm.html. This page contains more information about current snakemake cluster configuration. If not, it would be nice to be more specific with versioning: this tutorial works for snakemake 7.X.X but not for snakemake 8.X.X.

Reply:
In this current version, we wrote this issue as a note. although we don't recommend slurm, nice to test especially hyperqueue executor syntax with 8.X.X. version of snakemake

Reply:
when workflows run at scale, especially with lot of small jobs, would cause some extra overload on slurm cluster (actually on slurm accounting DB). That is why, we want to avoid slurm and use HQ if possible.  One can of course carefully write local rules in snakemake script to minimise these small jobs.